docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,448 @@
+The following Clusters will be affected by the changes:
+ ＊ default/example-cluster
+
+Changes for Cluster "default/example-cluster": 
+
+  NAMESPACE  KIND                   NAME                                  ACTION    
+  default    DockerCluster          example-cluster-7t7pl                 modified  
+  default    DockerMachineTemplate  example-cluster-control-plane-lt6kw   modified  
+  default    DockerMachineTemplate  example-cluster-md-0-infra-cjxs4      modified  
+  default    KubeadmConfigTemplate  example-cluster-md-0-bootstrap-m9sg8  modified  
+  default    KubeadmControlPlane    example-cluster-l7kx8                 modified  
+
+Modified objects are written to directory "output/modified"
+```
+
+Instead, if the command detects that the rebase operation will lead to a non-functional cluster (ClusterClasses are incompatible), the output will be similar to:
+```shell
+Detected a cluster with Cluster API installed. Will use it to fetch missing objects.
+Error: failed defaulting and validation on input objects: failed to run defaulting and validation on Clusters: failed validation of cluster.x-k8s.io/v1beta1, Kind=Cluster default/example-cluster: Cluster.cluster.x-k8s.io "example-cluster" is invalid: spec.topology.workers.machineDeployments[0].class: Invalid value: "default-worker": MachineDeploymentClass with name "default-worker" does not exist in ClusterClass "another-cluster-class"
+```
+In this example rebasing will lead to a non-functional Cluster because the ClusterClass is missing a worker class that is used by the Cluster.
+
+### Testing the affects of changing a ClusterClass
+
+When planning for a change on a ClusterClass you might want to understand what effects the change will have on existing clusters.
+
+```shell
+clusterctl alpha topology plan -f modified-first-cluster-class.yaml -o output/
+```
+When multiple clusters are affected, only the list of Clusters and ClusterClasses is presented.
+```shell
+Detected a cluster with Cluster API installed. Will use it to fetch missing objects.
+The following ClusterClasses will be affected by the changes:
+ ＊ default/first-cluster-class
+
+The following Clusters will be affected by the changes:
+ ＊ default/first-cluster
+ ＊ default/second-cluster
+
+No target cluster identified. Use --cluster to specify a target cluster to get detailed changes.
+```
+
+To get the full list of changes for the "first-cluster":
+```shell
+clusterctl alpha topology plan -f modified-first-cluster-class.yaml -o output/ -c "first-cluster"
+```
+Output will be similar to the full summary output provided in other examples.
+
+## Reference
+
+The topology plan operation is composed of the following steps:
+* Set the namespace on objects in the input with missing namespace.
+* Run the Defaulting and Validation webhooks on the Cluster and ClusterClass objects in the input.
+* Dry run the topology reconciler on the target cluster.
+* Capture all changes observed during reconciliation.
```

Comment:
This seems to be a bit out of place directly below Reference without any headline / context.

I would suggest to add a headline like "### How does topology plan work"